src/main/groovy/liquibase/harness/util/TestUtils.groovy

```diff
@@ -89,33 +41,60 @@ class TestUtils {
+     * @param script
      * @return
      */
-    static String cleanSql(String sql) {
-        if (sql == null) {
-            return null
+    static parseQuery(String script) {
+        if (script) {
+            script.replaceAll(/(?m)^--.*/, "") //remove comments
+                    .replaceAll(/(?m)^CREATE TABLE .*\w*.*DATABASECHANGELOG.*/, "") //remove create table queries for databasechangelog* tables
+                    .replaceAll(/(?m)^CREATE TABLE .*\w*.*databasechangelog.*/, "")
+                    .replaceAll(/(?m)^INSERT INTO .*\w*.*DATABASECHANGELOG.*/, "") //remove insert queries for databasechangelog* tables
+                    .replaceAll(/(?m)^INSERT INTO .*\w*.*databasechangelog.*/, "")
+                    .replaceAll(/(?m)^UPDATE .*\w*.*DATABASECHANGELOG.*/, "") //remove update queries for databasechangelog* tables
+                    .replaceAll(/(?m)^UPDATE .*\w*.*databasechangelog.*/, "")
+                    .replaceAll(/(?m)^DELETE FROM .*\w*.*databasechangelog.*/, "") //remove delete queries for databasechangelog* tables
+                    .replaceAll(/(?m)^DELETE FROM .*\w*.*DATABASECHANGELOG.*/, "")
+                    .replaceAll(/(?m)^SET SEARCH_PATH.*/, "") //specific replacement for Postgres
+                    .replaceAll(/\b(?:GO|USE lbcat)\b/, "") //specific replacement for MSSQL
+                    .replaceAll("(?m);\$", "") // remove semicolon
+                    .replaceAll(/^(?:[\t ]*(?:\r?\n|\r))+/, "") //remove empty lines
+                    .replaceAll(/(?m)^\s+/, "") //remove beginning whitespaces per line
+                    .replaceAll(/(?m)\s+$/, "") //remove trailing whitespaces per line
+                    .replaceAll("\r", "")
         }
-        return StringUtil.trimToNull(sql.replace("\r", "")
-                .replaceAll(/(?m)^--.*/, "") //remove comments
-                .replaceAll(/(?m)^\s+/, "") //remove beginning whitepace per line
-                .replaceAll(/(?m)\s+$/, "") //remove trailing whitespace per line
-        ) //remove trailing whitespace per line
     }
 
-    private static List<String> toSql(ChangeSet changeSet, Database db) {
-        return toSql(changeSet.changes, db)
-    }
-
-    private static List<String> toSql(List<? extends Change> changes, Database db) {
-        List<String> stringList = new ArrayList<>()
-        changes.each { stringList.addAll(toSql(it, db)) }
-        return stringList
+    static OutputStream executeCommandScope(String commandName, Map<String, Object> arguments) {
+        def commandScope = new CommandScope(commandName)
+        def outputStream = new ByteArrayOutputStream()
+        for (Map.Entry<String, Object> entry : arguments) {
+            commandScope.addArgumentValue(entry.getKey(), entry.getValue())
+        }
+        commandScope.setOutput(outputStream)
+        try {
+            commandScope.execute()
+        } catch (Throwable throwable) {
```

Comment:
In most cases it's bad practice to catch Throwable, If it' not absolutely needed change to Exception, otherwise add comment why do we need to catch Throwable

Reply:
++